### PR TITLE
Check for numpy type when converting metadata

### PR DESCRIPTION
**Pull request by mlincett**
Tentative fix for #18

### CODE REVIEW

File: skyreader/result.py [hunk 1]
```diff
@@ -537,7 +537,13 @@ def serialize(self) -> PyDictResult:
             for key in nside_data.dtype.metadata:
                 # dtype.metadata is a mappingproxy (dict-like) containing numpy-typed values
                 # convert numpy types to python bultins to be JSON-friendly
-                pydict[nside]['metadata'][key] = nside_data.dtype.metadata[key].item()
+                val = nside_data.dtype.metadata[key]
+                if type(val).__module__ == np.__name__ :
```
Comment (ric-evans) [inline skyreader/result.py:541]: Is there a way to do this with `isinstance()`?

**Review by ric-evans — CHANGES_REQUESTED** [review]

File: skyreader/result.py (hunk 1 above)
Comment (mlincett) [reply skyreader/result.py:541]: I would need to assume the data type(s) beforehand, or loop over all the possible data types to find a match.

An alternative would be to use `try / except AttributeError` but I guess exceptions should not be used as surrogates for conditional statements.

Comment (ric-evans) [reply skyreader/result.py:541]: Is there a common numpy base class for all its special data types?

Comment (ric-evans) [reply skyreader/result.py:541]: can you use https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.number?

Comment (mlincett) [reply skyreader/result.py:541]: This would only catch numeric types. I believe `numpy` may wrap strings and dates as well?

Comment (mlincett) [reply skyreader/result.py]: Alright, with `isinstance(val, np.generic)` it works!

**Pull request merged**